sched/sched/sched_addreadytorun.c

```diff
@@ -114,170 +114,170 @@ bool nxsched_add_readytorun(FAR struct tcb_s *btcb)
-                }
-              else
-                {
-                  nxsched_add_prioritized(btcb, &g_readytorun);
-                  btcb->task_state = TSTATE_TASK_READYTORUN;
-                }
+              rtcb->task_state = TSTATE_TASK_ASSIGNED;
             }
 
-          return false;
+          dq_addfirst((FAR dq_entry_t *)btcb, tasklist);
+          up_update_task(btcb);
+
+          btcb->cpu = cpu;
+          btcb->task_state = TSTATE_TASK_RUNNING;
+          ret = true;
+          break;
         }
+    }
+
+  return ret;
+}
 
-      tasklist = &g_assignedtasks[cpu];
+/****************************************************************************
+ * Name:  nxsched_add_readytorun
+ *
+ * Description:
+ *   This function adds a TCB to one of the ready to run lists. The list
+ *   will be:
+ *
+ *   1. The g_readytorun list if the task is ready-to-run but not running
+ *      and not assigned to a CPU.
+ *   2. The g_assignedtask[cpu] list if the task is running or if has been
+ *      assigned to a CPU.
+ *
+ *   If the currently active task has preemption disabled and the new TCB
+ *   would cause this task to be preempted, the new task is added to the
+ *   g_pendingtasks list instead.  The pending tasks will be made
+ *   ready-to-run when preemption isunlocked.
+ *
+ * Input Parameters:
+ *   btcb - Points to the blocked TCB that is ready-to-run
+ *
+ * Returned Value:
+ *   true if the currently active task (the head of the ready-to-run list)
+ *   has changed.
+ *
+ * Assumptions:
+ * - The caller has established a critical section before calling this
+ *   function (calling sched_lock() first is NOT a good idea -- use
+ *   enter_critical_section()).
+ * - The caller has already removed the input rtcb from whatever list it
+ *   was in.
+ * - The caller handles the condition that occurs if the head of the
+ *   ready-to-run list has changed.
+ *
+ ****************************************************************************/
 
-      /* Change "head" from TSTATE_TASK_RUNNING to TSTATE_TASK_ASSIGNED */
+bool nxsched_add_readytorun(FAR struct tcb_s *btcb)
+{
+  bool doswitch = false;
 
-      headtcb = (FAR struct tcb_s *)tasklist->head;
-      DEBUGASSERT(headtcb->task_state == TSTATE_TASK_RUNNING);
-      headtcb->task_state = TSTATE_TASK_ASSIGNED;
+  /* Add the btcb to the ready to run list, and
+   * try to run it on some CPU if it was added to the head of the list
+   */
 
-      /* Add btcb to the head of the g_assignedtasks
-       * task list and mark it as running
-       */
+  btcb->task_state = TSTATE_TASK_READYTORUN;
 
-      dq_addfirst_nonempty((FAR dq_entry_t *)btcb, tasklist);
-      up_update_task(btcb);
+  nxsched_add_prioritized(btcb, list_readytorun());
 
-      DEBUGASSERT(task_state == TSTATE_TASK_RUNNING);
-      btcb->cpu        = cpu;
-      btcb->task_state = TSTATE_TASK_RUNNING;
+  int target_cpu = btcb->flags & TCB_FLAG_CPU_LOCKED ? btcb->cpu :
```

Comment:
move before line 262 to avoid the variable definition in the middle of function

Reply:
Yes, thanks. I also improved the comment a bit